### ISSUE TITLE
Accessibility testing of docs examples would improve demo quality

### ISSUE THREAD

**Issue by marcysutton** [angular/material#7918]
Implementing an automated accessibility checker would improve the odds for accessibility of docs demos. Here is an example of a simple accessibility issue I found looking at the docs, which could be caught as an error with an API like axe-core:
```
<md-tabs md-selected="selectedIndex">
  <img ng-src="img/angular.png" class="centered">
  <md-tab ng-repeat="tab in tabs | orderBy:predicate:reversed" md-on-select="onTabSelected(tab)" md-on-deselect="announceDeselected(tab)" ng-disabled="tab.disabled">
    <md-tab-label>
      {{tab.title}}
      <img src="img/removeTab.png" ng-click="removeTab(tab)" class="delete">
    </md-tab-label>
    <md-tab-body>
      {{tab.content}}
    </md-tab-body>
  </md-tab>
</md-tabs>
```

In the above example, there are two images without alt attributes. We can easily fix this one API document. However, I suspect there are probably others that could be found programmatically as demos are tested and updated by warning the developer of accessibility issues with their code. 

I brought this up internally with the Material team, this issue serves as a record of this idea! We could integrate the open source [axe-core](https://github.com/dequelabs/axe-core) API into the build infrastructure, with the only cost being developer time to implement. Specifics TBD, I will talk about it more with @topherfangio.

**Comment by marcysutton** [comment]
Capturing comments from Slack: 

We’ll have to identify the appropriate places–the site has a lot going on with code in different contexts. Here are a few places I found that would benefit from testing: 
- Raw DOM snippets in the API pages
- Compiled demos, which have added semantics and keyboard support for custom directives
- The docs site itself (are there tests for that yet?)

The benefit being that it moves the work of auditing and checking for accessibility periodically by someone like me to a sustainable, automated tool in the development workflow.

**Comment by ThomasBurleson** [comment]
This issue is closed as part of our ‘Surge Focus on Material 2' efforts.
      For details, see our forum posting @ http://bit.ly/1UhZyWs.
